lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -348,20 +354,77 @@ def keyspace(self, query: Query) -> JSONOBJECT:
                     self.accumulate(aggregate, m.key, m.value)
         return aggregate
 
-    def datasets(self, request: Request, json: JSONOBJECT, query: Query) -> JSONOBJECT:
+    def datasets(
+        self, request: Request, aliases: dict[str, Any], json: JSONOBJECT, query: Query
+    ) -> JSONOBJECT:
         """Gather and paginate the selected datasets
 
         Run the query we've compiled, with pagination limits applied; collect
         results into a list of JSON objects including selected metadata keys.
 
         Args:
             request: The HTTP Request object
+            aliases: Map of join column aliases for each Metadata namespace
             json: The JSON query parameters
             query: The basic filtered SQLAlchemy query object
 
         Returns:
             The paginated dataset listing
         """
+
+        # Process a possible list of sort terms. By default, we sort by the
+        # dataset resource_id.
+        sorters = []
+        for sort in json.get("sort", ["dataset.resource_id"]):
+            if ":" not in sort:
+                k = sort
+                order = asc
+            else:
+                try:
+                    k, o = sort.split(":", maxsplit=1)
+                except ValueError:
+                    raise APIAbort(
+                        HTTPStatus.BAD_REQUEST,
+                        f"sort {sort!r} must have the form 'k[:o]'",
+                    )
+                if o.lower() == "asc":
+                    order = asc
+                elif o.lower() == "desc":
+                    order = desc
+                else:
+                    raise APIAbort(
+                        HTTPStatus.BAD_REQUEST,
+                        f"sort order in {sort!r} must be 'asc' or 'desc'",
+                    )
```

Comment:
Returning all of `sort` is good for context, but it would be better if we reported out the value of `o` (perhaps in addition), since that's what we are actually objecting to.

Reply:
But `o` is part of `sort`, and including the full value identifies the key and the sort keyword together, which seemed simpler and likely more immediately recognizable than reporting the separated values...

More importantly, though, this reminds me that I completely neglected to write error test cases. 😦

Reply:
> But `o` is part of `sort`, and including the full value identifies the key and the sort keyword together

True; but it doesn't quite illuminate why the code objected -- that is, the reader has to extrapolate from the value of `sort` to the actual keyword, and the reader might not necessarily reach the same conclusion that the code did....  🦐

Reply:
I don't see what in `The sort order in 'dataset.name:backwards' must be 'asc' or 'desc'` is unclear about the context. 😦 But, sure, I can make it wordier and possibly easier to miss by breaking them up since this seems to be really bothering you.